java/ql/src/Security/CWE/CWE-078/ExecTaintedEnvironment.qhelp

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>Passing unvalidated user input into the environment variables of a subprocess can allow an attacker to execute malicious code.</p>
+</overview>
+
+<recommendation>
+<p>If possible, use hard-coded string literals to specify the environment variable or its value. 
+Instead of passing the user input directly to the
+process or library function, examine the user input and then choose
+among hard-coded string literals.</p>
+
+<p>If the applicable environment variables cannot be determined at
+compile time, then add code to verify that the user input string is
+safe before using it.</p>
+</recommendation>
+
+<example>
+<p>In the following (BAD) example, the environment variable <code>PATH</code> is set to the value of the user input <code>path</code> without validation.</p>
+
+<sample src="ExecTaintedEnvironmentValue.java" />
+
+<p>In the following (BAD) example, an environment variable is set with a name that is derived from the user input <code>var</code> without validation.</p>
+
+<sample src="ExecTaintedEnvironmentName.java" />
+<p>In the following (GOOD) example, the user's input is validated before being used to set the environment variable.</p>
+
+<sample src="ExecTaintedEnvironmentValidated.java" />
+
+<p>In the following (GOOD) example, the user's input is checked and used to determine an environment variable to add.</p>
+
+<sample src="ExecTaintedEnvironmentChecked.java" />
+</example>
```

Comment:
I recommend adding a reference section with at least one reference. This will ensure that when we publish the docs, they will automatically include a link to CWE 78.

Potentially one of the following might be suitable? But you're probably better placed than me to find a good reference 😁 

- [ENV02-J. Do not trust the values of environment variables - SEI CERT Oracle Coding Standard for Java - Confluence](https://wiki.sei.cmu.edu/confluence/display/java/ENV02-J.+Do+not+trust+the+values+of+environment+variables)
- [Environment Variables (The Java™ Tutorials > Essential Java Classes > The Platform Environment)](https://docs.oracle.com/javase/tutorial/essential/environment/env.html)

For formatting information, see: https://github.com/github/codeql/blob/main/docs/query-help-style-guide.md#websites

Reply:
Hi @felicitymay! Thank you for reviewing this.

I have added a few references.

> ENV02-J. Do not trust the values ...

Thank you for pointing out CMU's SEI docs. I've now found a different page through them which might be a bit more in-line with the focus of this query, but it's from their C/C++ guide, so I'm not sure if it's appropriate to add to a Java query. I will consult with my team.

Thank you!